weatherman.py

```diff
@@ -0,0 +1,178 @@
+import argparse
+import calendar
+from datetime import datetime
+import os
+
+RED = "\033[31m"
+BLUE = "\033[34m"
+RESET = "\033[0m"
+
+
+class WeatherReading:
+    def __init__(self, observation_date, max_temp, min_temp, max_humidity, mean_humidity):
+        self.date = observation_date
+        self.max_temp = max_temp
+        self.min_temp = min_temp
+        self.max_humidity = max_humidity
+        self.mean_humidity = mean_humidity
+                                                                                                   
+    def __repr__(self):
+        return (f"WeatherReading(date={self.date}, max_temp={self.max_temp}, min_temp={self.min_temp},"
+                f"max_humidity={self.max_humidity}, mean_humidity={self.mean_humidity})")
+
+
+def parse_weather_file(file_path):
```

Comment:
If you opted to go for classes, everything should be in that format. All classes should comply with all OOP principles and they should be stored in separate files.